pkg/ucp/frontend/controller/resourcegroups/util.go

```diff
@@ -59,38 +61,161 @@ func (e *InvalidError) Is(err error) bool {
+		_, err = store.GetResource[datamodel.ResourceType](ctx, client, resourceTypeID.String())
 		if errors.Is(err, &store.ErrNotFound{}) {
-			return nil, &NotFoundError{Message: fmt.Sprintf("resource group %q not found", resourceGroupID.String())}
+
+			// Return the error as-is to fallback to the legacy routing behavior.
+			return nil, err
+
+			// Uncomment this when we remove the legacy routing behavior.
+			// return nil, &InvalidError{Message: fmt.Sprintf("resource type %q not found", id.Type())}
 		} else if err != nil {
-			return nil, fmt.Errorf("failed to find resource group %q: %w", resourceGroupID.String(), err)
+			return nil, fmt.Errorf("failed to find resource type %q: %w", id.Type(), err)
+		}
+	}
+
+	locationID, err := datamodel.ResourceProviderLocationIDFromResourceID(id, locationName)
+	if err != nil {
+		return nil, err
+	}
+
+	location, err := store.GetResource[datamodel.Location](ctx, client, locationID.String())
+	if errors.Is(err, &store.ErrNotFound{}) {
+
+		// Return the error as-is to fallback to the legacy routing behavior.
+		return nil, err
+
+		// Uncomment this when we remove the legacy routing behavior.
+		// return nil, &InvalidError{Message: fmt.Sprintf("location %q not found for resource provider %q", locationName, id.ProviderNamespace())}
+	} else if err != nil {
+		return nil, fmt.Errorf("failed to find location %q: %w", locationID.String(), err)
+	}
+
+	// Check if the location supports the resource type.
+	// Resource types are case-intensitive so we have to iterate.
+	var locationResourceType *datamodel.LocationResourceTypeConfiguration
+
+	// We special-case two pseudo-resource types: "locations/operationstatuses" and "locations/operationresults".
+	// If the resource type is one of these, we can return the downstream URL directly.
+	if isOperationResourceType(id) {
+		locationResourceType = &datamodel.LocationResourceTypeConfiguration{
+			APIVersions: map[string]datamodel.LocationAPIVersionConfiguration{
+				apiVersion: {}, // Assume this API version is supported.
+			},
 		}
+	} else {
+		// Ex: Applications.Test/testResources => testResources
+		search := strings.TrimPrefix(strings.ToLower(id.Type()), strings.ToLower(id.ProviderNamespace())+resources.SegmentSeparator)
+		for name, rt := range location.Properties.ResourceTypes {
+			if strings.EqualFold(name, search) {
+				copy := rt
+				locationResourceType = &copy
+				break
+			}
+		}
+	}
+
+	if locationResourceType == nil {
+		return nil, &InvalidError{Message: fmt.Sprintf("resource type %q not supported by location %q", id.Type(), locationName)}
+	}
+
+	// Now check if the location supports the resource type. If it does, we can return the downstream URL.
+	_, ok := locationResourceType.APIVersions[apiVersion]
+	if !ok {
+		return nil, &InvalidError{Message: fmt.Sprintf("api version %q is not supported for resource type %q by location %q", apiVersion, id.Type(), locationName)}
+	}
+
+	// If we get here, the we're all good.
```

Comment:
nit: If we get **to** here, **then** we're all good

Reply:
:)